### PR TITLE
MDEXP-265 Applying mapping profile  - holdings and item transformatio…

### PR DESCRIPTION
**Pull request by illia-borysenko**
Will provide details later

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/service/manager/export/ExportManagerImpl.java [hunk 1]
```diff
@@ -199,9 +246,8 @@ private ExportResult getExportResult(AsyncResult<Object> asyncResult, ExportPayl
       errorLogService.saveGeneralError(ErrorCode.GENERIC_ERROR_CODE.getDescription(), exportPayload.getJobExecutionId(), exportPayload.getOkapiConnectionParams().getTenantId());
       return ExportResult.failed(ErrorCode.GENERIC_ERROR_CODE);
     } else {
-      LOGGER.info("Export batch has successfully completed");
+      LOGGER.info("Export has been successfully passed");
```
[the remark this answers is not in the document]
Comment (illia-borysenko) [reply src/main/java/org/folio/service/manager/export/ExportManagerImpl.java:249]: seems some incorrect changes applied after resolving conflicts, I will check all similar places

**Review by andrii-novytskyi — APPROVED** [review]

File: src/main/java/org/folio/service/manager/export/ExportManagerImpl.java [hunk 2]
```diff
@@ -87,48 +93,89 @@ public void exportData(JsonObject request) {
...
+        && mappingProfile.getRecordTypes().contains(RecordType.SRS)
+        && !mappingProfile.getRecordTypes().contains(RecordType.INSTANCE)) {
+      SrsLoadResult srsLoadResult = loadSrsMarcRecordsInPartitions(identifiers, exportPayload.getJobExecutionId(), params);
+      LOGGER.info("Records that are not present in SRS: {}", srsLoadResult.getInstanceIdsWithoutSrs());
       List<String> marcToExport = srsRecordService.transformSrsRecords(mappingProfile, srsLoadResult.getUnderlyingMarcRecords(),
         exportPayload.getJobExecutionId(), params);
       exportService.exportSrsRecord(marcToExport, fileExportDefinition);
       LOGGER.info("Number of instances not found in SRS: {}", srsLoadResult.getInstanceIdsWithoutSrs().size());
+      mappingProfileService
+          .getById(DEFAULT_MAPPING_PROFILE_ID, params.getTenantId())
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/manager/export/ExportManagerImpl.java:112]: No need to store DEFAULT_MAPPING_PROFILE_ID in the ExportManager, just create a method in mappingProfileService to retrieve default profile

File: src/main/java/org/folio/service/manager/export/ExportManagerImpl.java [hunk 3]
```diff
@@ -87,48 +93,89 @@ public void exportData(JsonObject request) {
...
+                generateRecordsOnTheFly(
+                    exportPayload,
+                    identifiers,
+                    fileExportDefinition,
+                    defaultMappingProfile,
+                    params,
+                    srsLoadResult);
+                blockingPromise.complete();
+              })
+          .onFailure(
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/manager/export/ExportManagerImpl.java:124]: I don't like this manner of code formatting, let's revert it back to the usual version, and discuss, it with all the backend people

Comment (Igor-Gorchakov) [reply src/main/java/org/folio/service/manager/export/ExportManagerImpl.java:124]: This makes the async code completely unreadable for me

File: src/main/java/org/folio/service/manager/export/ExportManagerImpl.java [hunk 4]
```diff
@@ -44,6 +48,7 @@
   private static final int POOL_SIZE = 2;
   private static final int SRS_LOAD_PARTITION_SIZE = 50;
   private static final int INVENTORY_LOAD_PARTITION_SIZE = 50;
+  private static final String DEFAULT_MAPPING_PROFILE_ID = "25d81cbe-9686-11ea-bb37-0242ac130002";
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/manager/export/ExportManagerImpl.java:51]: No need to store DEFAULT_MAPPING_PROFILE_ID in the ExportManager, just create a method `getDefaultProfile` in the MappingProfileService to retrieve the default profile, store DEFAULT_MAPPING_PROFILE_ID there

**Review by Igor-Gorchakov — CHANGES_REQUESTED** [review]

**Review by Igor-Gorchakov — APPROVED** [review]

File: src/main/java/org/folio/service/mapping/RuleFactory.java [hunk 5]
```diff
@@ -80,7 +80,8 @@
       } else if (isInstanceTransformationValidAndBlank(mappingTransformation)) {
         rule = createDefaultByTransformations(mappingTransformation, defaultRules);
       } else if(HOLDINGS.equals(mappingTransformation.getRecordType()) || ITEM.equals(mappingTransformation.getRecordType()) ) {
-        LOGGER.error("Transformation with empty value is not available for record types Holdings and Item");
+        LOGGER.error(String.format("No transformation provided for field name: %s, and with record type: %s",
```
Comment (andrii-novytskyi) [inline src/main/java/org/folio/service/mapping/RuleFactory.java:83]: Please, a part from logger, save ErrorLog here

**Pull request merged**